src/inet/routing/leach/LeachBS.cc

```diff
@@ -0,0 +1,103 @@
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+//
+
+#include "inet/common/IProtocolRegistrationListener.h"
+#include "inet/common/ModuleAccess.h"
+#include "inet/common/ProtocolTag_m.h"
+#include "inet/linklayer/common/InterfaceTag_m.h"
+#include "inet/networklayer/common/L3AddressTag_m.h"
+#include "inet/physicallayer/wireless/common/contract/packetlevel/SignalTag_m.h"
+#include "inet/routing/leach/LeachBS.h"
+#include "inet/routing/leach/Leach.h"
+
+using namespace std;
+
+namespace inet {
+namespace leach {
+
+Define_Module(LeachBS);
+
+LeachBS::LeachBS() {
+
+}
+
+LeachBS::~LeachBS() {
+    // TODO Auto-generated destructor stub
+}
+
+void LeachBS::initialize(int stage) {
+    RoutingProtocolBase::initialize(stage);
+
+    //reads from omnetpp.ini
+    if (stage == INITSTAGE_LOCAL) {
+        sequencenumber = 0;
+        host = getContainingNode(this);
+        interfaceTable.reference(this, "interfaceTableModule", true);
+
+        bsPktReceived = 0;
+
+    } else if (stage == INITSTAGE_ROUTING_PROTOCOLS) {
+        registerService(Protocol::manet, nullptr, gate("ipIn"));
+        registerProtocol(Protocol::manet, gate("ipOut"), nullptr);
+    }
+}
+
+void LeachBS::start() {
+    /* Search the 802154 interface */
+    configureInterfaces();
+
+}
+
+void LeachBS::stop() {
+
+}
+
+void LeachBS::configureInterfaces() {
+    int numInterfaces = interfaceTable->getNumInterfaces();
+    for (int i = 0; i < numInterfaces; i++) {
```

Comment:
This just pick the fors wireless interface. an explicit `interface` parameter would be better with wlan0 as a default.

Reply:
Resolved in https://github.com/inet-framework/inet/pull/929/commits/8dcc7d590e6952bca4f714c7611c13a4928a3161